src/components/organisms/EntityList/EntityCreationModal/TestCreationForm/TestCreationForm.tsx

```diff
@@ -0,0 +1,180 @@
+  onFail: (err: any) => void;
+  testSources: SourceWithRepository[];
+  executors: Executor[];
+};
+
+const additionalFields: {[key: string]: React.FC<any>} = {
+  git: GitCreationFormFields,
+  'file-uri': FileContentFields,
+  custom: CustomCreationFormFields,
+  string: StringContentFields,
+};
+
+const TestCreationForm: React.FC<TestCreationFormProps> = props => {
+  const {form, testSources, executors, onSuccess, onFail} = props;
+
+  const remappedExecutors = remapExecutors(executors);
+  const remappedCustomTestSources = remapTestSources(testSources);
+
+  const [localLabels, setLocalLabels] = useState<readonly Option[]>([]);
+
+  const [addTest, {isLoading}] = useAddTestMutation();
+
+  const onSave = (values: any) => {
+    const {testSource, testType} = values;
+
+    const requestBody = {
+      name: values.name,
+      type: testType,
+      labels: decomposeLabels(localLabels),
+      content: getSourcePayload(values, testSources),
+      ...getCustomSourceField(testSource),
+    };
+
+    return addTest(requestBody)
+      .then(res => {
+        onSuccess(res);
+      })
+      .catch(err => {
+        onFail(err);
+      });
+  };
+
+  const selectedExecutor = useMemo(() => {
+    return executors.find((executor: Executor) => executor.executor?.types?.includes(form.getFieldValue('testType')));
+  }, [form.getFieldValue('testType')]);
+
+  return (
+    <Form form={form} layout="vertical" name="test-creation" onFinish={onSave} style={{flex: 1}} labelAlign="right">
+      <StyledFormSpace size={24} direction="vertical">
+        <Text className="regular big">Test details</Text>
+        <FormItem
+          name="name"
+          label="Name"
+          rules={[required, k8sResourceNamePattern, k8sResourceNameMaxLength]}
+          required
+        >
+          <Input placeholder="e.g.: my-test" />
+        </FormItem>
+        <FormItem name="testType" label="Type" rules={[required]} required>
+          <Select placeholder="Select from available executors..." showSearch options={remappedExecutors} />
+        </FormItem>
+        <FormItem
+          noStyle
+          shouldUpdate={(prevValues, currentValues) => prevValues.testSource !== currentValues.testSource}
+        >
+          {() => {
+            if (!selectedExecutor) {
+              return null;
+            }
+
+            const options = [
+              ...remappedCustomTestSources,
+              ...testSourceBaseOptions.filter(
+                option =>
+                  selectedExecutor.executor?.contentTypes?.includes(String(option.value)) ||
+                  !selectedExecutor.executor?.contentTypes?.length
+              ),
+            ];
+
+            return (
+              <FormItem rules={[required]} label="Source" name="testSource" required>
+                <Select placeholder="Select a source..." options={options} showSearch />
+              </FormItem>
+            );
+          }}
+        </FormItem>
```

Comment:
I see that there is no `Branch/Commit`, which is in the edition. Should it be there?

Reply:
Nope, for test creation it is only a branch. Possibility to select input a commit only available in edition